src/main/java/dev/codesupport/web/common/service/http/RestRequest.java

```diff
@@ -0,0 +1,71 @@
+package dev.codesupport.web.common.service.http;
+
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+@EqualsAndHashCode
+@ToString
+abstract class RestRequest<S extends Serializable, T extends Serializable> {
+
+    protected String url;
+    protected Map<String, String> headers;
+    protected Map<String, String> parameters;
+    protected S requestObject;
+    protected Class<T> responseClass;
+    protected Class<S> requestClass;
+
+    RestRequest() {
+        this.headers = new HashMap<>();
+        this.parameters = new HashMap<>();
+        this.headers.put("User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.11 (KHTML, like Gecko) Chrome/23.0.1271.95 Safari/537.11");
```

Comment:
What's this for?

Reply:
So the discord app thinks you're being rude if you don't tell it what you are, and it will not respond to you if you don't.  This is actually fairly common in APIs from my experience, it's basically telling it if you are a chrome browser or something else, though it doesn't really seem to care what you tell it in this case.  We can adjust this header if a more appropriate one is found, this is one I found by seeing what other people used when hitting the discord API.

Reply:
Makes sense, not sure if it's worth being moved to a config?

Reply:
we could, depends how important it really is, I can't think of a single situation where we would need to change it, it just needs to be there, but I can go either way

Reply:
I feel like it would be easier to be able to quickly reference it from a config rather than having to type it out?

Reply:
You mean if we wanted to change it?  We're never going to need to change it, unless it actually gets deprecated as google is trying to do, in which case we would just delete it.